src/core_modules/capture-core/components/Pages/Search/SearchPage.container.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import { connect } from 'react-redux';
+import { SearchPage as SearchPageComponent } from './SearchPage.component';
+import { programCollection } from '../../../metaDataMemoryStores';
+import { TrackerProgram } from '../../../metaData/Program';
+import { withErrorMessageHandler, withLoadingIndicator } from '../../../HOC';
+import type { PropsFromRedux } from './SearchPage.types';
+
+const mapStateToProps = (state: ReduxState): PropsFromRedux => {
+    const { currentSelections, activePage } = state;
+
+    const trackedEntityTypesWithCorrelatedPrograms =
+      [...programCollection.values()]
+          .filter(program => program instanceof TrackerProgram)
+          .reduce((acc, {
+              id: programId,
+              name: programName,
+              trackedEntityType: { id: trackedEntityTypeId, name: trackedEntityTypeName },
```

Comment:
I am getting a flow error here that I dont like much. Can you please help me fix it @JoakimSM?

Reply:
Hmm, flow doesn't pick up the result from the filter function always being a tracker program. This is a recurring problem, i.e. I don't have a good solution for now.

Reply:
Hm yeah. Strange `flow`. Can you explain to me -coz I dont get it- how is it that a `programmCollection` which is typed as ```Map<string,Program>``` can end up being `EventProgram` or `TrackerProgram`?

Reply:
`EventProgram` and `TrackerProgram` is a child class of `Program`. The collection contains programs of some kind.

Reply:
Got it thanks!